llvm/test/CodeGen/DirectX/ResourceAccess/load_typedbuffer.ll

```diff
@@ -30,7 +30,7 @@ define void @load_float4(i32 %index, i32 %elemindex) {
   ; CHECK: %[[LOAD:.*]] = call { <4 x float>, i1 } @llvm.dx.resource.load.typedbuffer.v4f32.tdx.TypedBuffer_v4f32_1_0_0t(target("dx.TypedBuffer", <4 x float>, 1, 0, 0) %buffer, i32 %index)
   ; CHECK: %[[VALUE:.*]] = extractvalue { <4 x float>, i1 } %[[LOAD]], 0
   ; CHECK: extractelement <4 x float> %[[VALUE]], i32 %elemindex
-  %dynamic = getelementptr inbounds <4 x float>, ptr %ptr, i32 0, i32 %elemindex
+  %dynamic = getelementptr inbounds float, ptr %ptr, i32 %elemindex
```

Comment:
So somehow this change allows gep instructions to only need to specify the scalar type?
I'm guessing this is line 184 or 102 in DxilResourceAccess.cpp making this change.

Reply:
This test was actually just incorrect, but the old code happened to be wrong in just the right way that it looked like it did the right thing. The test case is trying to load a single float dynamically, so why was it loading from a `<4 x float>`?

Consider this HLSL to see what I mean ([godbolt](https://hlsl.godbolt.org/z/or4cqohqP)):
```hlsl
RWBuffer<float4> Out;
Buffer<float4> In;

[numthreads(4,1,1)]
void main(uint GID : SV_GroupID) {
  Out[GID][GID] = In[GID][GID];
}
```

The load uses an extractelement in clang's output, but you can see the gep used by the store around line 35 - it loads a `float`, not a `<4 x float>`, as otherwise the indexing would be wrong.